todos-app/src/App/App.tsx

```diff
@@ -3,22 +3,32 @@ import NewTodoInput from "./NewTodoInput/NewTodoInput";
 import TodoList from "./TodoList/TodoList";
 import ListFooter from "./ListFooter/ListFooter";
 import { ITodo } from "../DataStructure";
-import setUUID from "../UUID";
 import "../App.css";
 
 const App: React.FC = () => {
   const [todos, setTodos] = useState<ITodo[]>([]);
 
   const addHandler = (input: ITodo) => {
-    setTodos((prev) => [input, ...prev]);
+    setTodos((prev) => [...prev, input]);
+  };
+
+  const toggleHandler = (id: string) => {
+    setTodos((prev) =>
+      prev.map((todo) => {
+        if (todo.id === id) {
+          todo.completed = !todo.completed;
+        }
+        return todo;
+      })
+    );
   };
 
   return (
     <section className="todoapp">
       <NewTodoInput onAdd={addHandler} />
       {todos.length ? (
         <>
-          <TodoList todos={todos} />
+          <TodoList todos={todos} onToggle={toggleHandler} />
```

Comment:
It was not obvious until I read the toggle function that it was toggling completed tasks. Perhaps this could use a better name...

Reply:
Rename it to toggleTaskHandler. Will that do?